recipes/enchant/all/patches/0003-fix-flock-usage-for-windows.patch

```diff
@@ -0,0 +1,30 @@
+--- src/pwl.c
++++ src/pwl.c
+@@ -59,8 +59,10 @@
+ #include <string.h>
+ #include <sys/types.h>
+ #include <sys/stat.h>
+-#include <sys/file.h>
+ #include <fcntl.h>
++#ifndef _WIN32
++# include <sys/file.h>
++#endif
+ 
+ #include <glib.h>
+ #include <glib/gstdio.h>
+@@ -182,8 +184,13 @@
+ 
+ static int edit_dist(const char* word1, const char* word2);
+ 
+-#define enchant_lock_file(f) flock (fileno (f), LOCK_EX)
+-#define enchant_unlock_file(f) flock (fileno (f), LOCK_UN)
++#ifdef HAVE_FLOCK
++# define enchant_lock_file(f) flock (fileno (f), LOCK_EX)
++# define enchant_unlock_file(f) flock (fileno (f), LOCK_UN)
++#else
++# define enchant_lock_file(f)
```

Comment:
I worry providing locks as no-ops may open a pandora's box on Windows, and Windows users will face deadlocks, race-conditions, etc. which are hard to debug.
perhaps, it's better to provide an actual implementation via [LockFileEx](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-lockfileex)?
or if it's too much, I would just disable Windows in the recipe, instead of providing potentially broken implementation.

Reply:
This was taken from https://github.com/winlibs/enchant/commit/df5a4586e09d67bf2c208c7dfd0517657d06910e#diff-a54665f3f5d38281ae6f02c4e410a88d332d5b9f42466f05c35246ea23cd201d  
Maybe @cmb69 could provide some more details as the commit message glosses over this.

Reply:
That was just a quick "fix" to work around the issue that `flock()` is not available on Windows. The proper fix might be to actually use the gnulib submodule and to adapt the VS solutions; or possibly to revert the relevant parts of https://github.com/AbiWord/enchant/commit/edf662a5ec8db85aa6e5a5d00531ab6b8773131f.

Reply:
Implemented a more robust cross-platform locking/unlocking in bcf5eca

Reply:
thanks, much better